compiler/util/files.cpp

```diff
@@ -638,6 +638,202 @@ void genIncludeCommandLineHeaders(FILE* outfile) {
   }
 }
 
+std::string genMakefileEnvCache(void) {
+  std::string result;
+  std::map<std::string, const char*>::iterator env;
+
+  for (env = envMap.begin(); env != envMap.end(); ++env) {
+    const std::string& key = env->first;
+    const char* oldPrefix = "CHPL_";
+    const char* newPrefix = "CHPL_MAKE_";
+    INT_ASSERT(key.substr(0, strlen(oldPrefix)) == oldPrefix);
+    std::string keySuffix = key.substr(strlen(oldPrefix), std::string::npos);
+    std::string chpl_make_key = newPrefix + keySuffix;
+    result += chpl_make_key + "=" + std::string(env->second) + "|";
+  }
+
+  return result;
+}
+
+//
+// TODO: Multi-Locale Interop prototype code!
+//
+// For a clean slate, branch here when generating a Makefile under MLI. We
+// can integrate these changes into the main function later.
+//
+void codegen_makefile_mli(fileinfo* mli_client, fileinfo* mli_server,
```

Comment:
There is a lot of code here that I know already happens in codegen_makefile.  Please refactor so that we share code instead of having to remember to update this function when making improvements to the original (which is a maintenance burden)

Reply:
I dreaded this.

Reply:
XD "It's come at last.  The day I knew would come at last has come at last."